pkg/registry/discovery/endpointslice/strategy.go

```diff
@@ -65,7 +65,7 @@ func (endpointSliceStrategy) PrepareForUpdate(ctx context.Context, obj, old runt
 	newEPS.ObjectMeta = v1.ObjectMeta{}
 	oldEPS.ObjectMeta = v1.ObjectMeta{}
 
-	if !apiequality.Semantic.DeepEqual(newEPS, oldEPS) {
+	if !apiequality.Semantic.DeepEqual(newEPS, oldEPS) || !apiequality.Semantic.DeepEqual(ogNewMeta.Labels, ogOldMeta.Labels) {
```

Comment:
I was actually suggesting going even step further and adding:
```
else {
  ogNewMeta.Generation = og.OldMeta.Generation
}
```

But I would like to hear Jordan`s opinion (if that doesn't violate some conventions, backward compatibility, etc.)

Reply:
rest.BeforeUpdate already resets generation to the current value before calling strategy.PrepareForUpdate:

https://github.com/kubernetes/kubernetes/blob/c8fe1d9c0699f8d1fd93f47f2a62a4c83ccf5b6f/staging/src/k8s.io/apiserver/pkg/registry/rest/update.go#L105-L113

Reply:
Good point - I missed that thanks.